sdk/monitor/monitor-opentelemetry-exporter/src/utils/spanUtils.ts

```diff
@@ -220,28 +287,18 @@ export function readableSpanToEnvelope(span: ReadableSpan, ikey: string): Envelo
       baseType = "RequestData";
       baseData = createRequestData(span);
       break;
-    case SpanKind.INTERNAL:
-      baseType = "RemoteDependencyData";
-      name = "Microsoft.ApplicationInsights.RemoteDependency";
-      baseData = createInProcData(span);
-      break;
     default:
       // never
       diag.error(`Unsupported span kind ${span.kind}`);
       throw new Error(`Unsupported span kind ${span.kind}`);
   }
 
-  if (span.attributes[AzNamespace] === MicrosoftEventHub) {
-    parseEventHubSpan(span, baseData);
-  } else if (span.attributes[AzNamespace]) {
-    switch (span.kind) {
-      case SpanKind.INTERNAL:
-        (baseData as RemoteDependencyData).type = `${INPROC} | ${span.attributes[AzNamespace]}`;
-        break;
-      default: // no op
-    }
+  if (span.kind === SpanKind.PRODUCER) {
+    baseData.type = DependencyTypes.QueueMessage;
+  }
+  if (span.kind === SpanKind.INTERNAL && span.parentSpanId) {
```

Comment:
Also I think `data.success=True` always for `Internal`

Reply:
Even if explicitly set to error?, that sounds weird